src/Jackett.Common/Indexers/AwesomeHD.cs

```diff
@@ -0,0 +1,188 @@
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Jackett.Common.Models;
+using Jackett.Common.Models.IndexerConfig;
+using Jackett.Common.Services.Interfaces;
+using Jackett.Common.Utils;
+using Newtonsoft.Json.Linq;
+using NLog;
+
+namespace Jackett.Common.Indexers
+{
+    public class AwesomeHD : BaseWebIndexer
+    {
+        private static string SearchUrl => "searchapi.php";
+        private static string TorrentUrl => "torrents.php";
+        private new ConfigurationDataPasskey configData => (ConfigurationDataPasskey)base.configData;
+
+        public AwesomeHD(IIndexerConfigurationService configService, Utils.Clients.WebClient c, Logger l, IProtectionService ps)
+            : base("Awesome-HD",
+                description: "An HD tracker",
+                link: "https://awesome-hd.me/",
+                caps: new TorznabCapabilities(),
+                configService: configService,
+                client: c,
+                logger: l,
+                p: ps,
+                configData: new ConfigurationDataPasskey())
+        {
+            Encoding = Encoding.UTF8;
+            Language = "en-us";
+            Type = "private";
+
+            TorznabCaps.SupportsImdbMovieSearch = true;
+            TorznabCaps.SupportsImdbTVSearch = true;
+
+            AddCategoryMapping(1, TorznabCatType.MoviesHD);
+            AddCategoryMapping(2, TorznabCatType.TVHD);
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            if (configData.Passkey.Value.Length != 32)
+                throw new Exception("Invalid Passkey configured. Expected length: 32");
+
+            var releases = await PerformQuery(new TorznabQuery());
+            await ConfigureIfOK(string.Empty, releases.Any(),
+                                () => throw new Exception("Could not find release from this URL."));
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var passkey = configData.Passkey.Value;
+
+            var qc = new NameValueCollection
+            {
+                {"passkey", passkey}
+            };
+
+            if (!string.IsNullOrWhiteSpace(query.ImdbID))
+            {
+                qc.Add("action", "imdbsearch");
+                qc.Add("imdb", query.ImdbID);
+            }
+            else if (!string.IsNullOrWhiteSpace(query.GetQueryString()))
+            {
+                qc.Add("action", "titlesearch");
+                qc.Add("title", query.SearchTerm); // not use query.GetQueryString(), see the season code below
+            }
+            else
+                qc.Add("action", "latestmovies");
+
```

Comment:
On the dashboard, a keyword-less search of this indexer with the category filter set to 5040(TV) returns nothing. I presume this is due to line 80.
The yaml allowed empty searchstr parm, does the API not allow empty title on titlesearch?
as it stands, Sonarr users setting up their AHD indexers for the first time, or try to test their indexer with the Sonarr test button, will fail the test as movie cats are not usually added to their Sonarr config.

Reply:
The API `latestmovies` only reruns movies and a keyword-less search in the `titlesearch` API raises an error.
I made a hack to make it working with Sonarr, but this has to be solved by the SYSOP. I added this problem in the first message.